### DISCUSSION TITLE
Create practice difficulty: send selected objects to new difficulty

### DISCUSSION THREAD

**Discussion by Doksha** [ppy/osu#19953] [Ideas] [answered]
Currently to create a practice difficulty, you have to create a new difficulty (without clearing objects), select the hit objects before and after the section you want to practice, delete them, save the difficulty, (ctrl + z, then repeat if you want to make multiple). 

It shouldn't be too hard for there to be a button added in the editor to take the currently selected circles and create a new difficulty with just them (this new difficulty would not be opened, the user would remain in the original difficulty)
This button could be in the context menu when you right click a group of hit objects (right below Delete)

**Comment by peppy** [comment, marked as answer]
This is not a great UX. We already have plans to add a proper practice mode UI so users can stop making these "practice beatmaps"

**Reply by frenzibyte** [reply]
Existing discussion threads for reference:
- https://github.com/ppy/osu/discussions/19932
- https://github.com/ppy/osu/discussions/16723
- https://github.com/ppy/osu/discussions/14629
- https://github.com/ppy/osu/discussions/13079
- https://github.com/ppy/osu/discussions/13076
- https://github.com/ppy/osu/discussions/13013

**Reply by Doksha** [reply]
TBH none of these ideas seem better (at least for practicing FL) than just downloading stable, opening stable, downloading the beatmap, creating a new practice difficulty, exporting it, then switching back to lazer; maybe they would work fine for when you want to just practice a single section of a map a couple times, but for practicing FL none of these ideas work, they just don't make sense for repeatability and seem like more of a hassle. (and "send to new defficulty" would be insanely easy to implement, hell I could probably do it and I have no idea what language lazer is in, but I dont need the feature often enough for the time investment to be worth it for me)

Simply put _This should not be a mod_, this should be an option in the editor. If you are really against adding something to the UI, just make it a hidden keybind (Something like Ctrl+Shift+Alt+S).

Unless there really is a completely different and somehow better idea (I really cant think of anything better than creating a practice difficulty, please enlighten me if there is something better) this should be implemented (and even if there is a better idea, this should still be implemented since its just so damn simple)
